src/main/java/net/openhft/chronicle/core/io/VanillaReferenceCounted.java

```diff
@@ -97,6 +96,10 @@ private boolean valueCompareAndSet(int from, int to) {
         return UnsafeMemory.INSTANCE.compareAndSwapInt(this, VALUE, from, to);
     }
 
+    private int valueGetAndSet(int to) {
+        return UnsafeMemory.INSTANCE.getAndSetInt(this, VALUE, to);
```

Comment:
I had to add this atomically get and set the value.

Reply:
What does this tell you that CAS doesn't?

Reply:
It allows us to atomically set the value and get the value that was before we set it, that way we can know if we were the ones responsible for changing the value from >=  1 to zero and therefore whether it's safe to call `onRelease`. We _could_ implement `getAndSet` with `compareAndSet` in a loop, but `getAndSet` means we don't have to do that.

Reply:
We can't force the release of the resource. If releaseLast is called while another thread still uses the resource, we can't have it SEG faulting.